src/routes/Articles/service.ts

```diff
@@ -0,0 +1,138 @@
+    return existingArticle;
+  }
+
+  private async validateAuth(articleUserId: number, userId: number, message: string) {
+    if (articleUserId !== userId) throw new ForbiddenException(message);
+  }
+
+  async postArticle(userId: number, createArticleDto: CreateArticleRequest) {
+    const articleEntity = await this.articleRepository.create(userId, createArticleDto);
+    return new Article({ ...articleEntity, isLiked: false });
+  }
+
+  async getArticleById(articleId: number, userId: number) {
+    const articleEntity = await this.articleRepository.findById(articleId);
+
+    if (!articleEntity) {
+      throw new NotFoundException(EXCEPTION_MESSAGES.articleNotFound);
+    }
+
+    const isLiked = await this.likeRepository.findIsLiked(articleId, userId);
+    return new Article({ ...articleEntity, isLiked });
+  }
+
+  async editArticle(articleId: number, userId: number, editArticleRequestDto: EditArticleRequest) {
+    const existingArticle = await this.getExistingArticle(articleId);
+    await this.validateAuth(existingArticle.userId, articleId, AUTH_MESSAGES.update);
+
+    const articleEntity = await this.articleRepository.update(articleId, editArticleRequestDto);
+    const isLiked = await this.likeRepository.findIsLiked(articleId, userId);
+    return new Article({ ...articleEntity, isLiked });
+  }
+
+  async deleteArticle(articleId: number, userId: number) {
+    const existingArticle = await this.getExistingArticle(articleId);
+    await this.validateAuth(existingArticle.userId, articleId, AUTH_MESSAGES.delete);
+    await this.articleRepository.delete(articleId);
+  }
+
+  async getArticleList(userId: number, getArticleListDto: GetArticleListRequest) {
+    const articleListResult = await this.articleRepository.getArticleList(getArticleListDto);
+
+    const articles = await Promise.all(
+      articleListResult.list.map(async (articleEntity) => {
+        if (!userId) return new Article({ ...articleEntity, isLiked: false });
+        const isLiked = await this.likeRepository.findIsLiked(articleEntity.id, userId);
+        return new Article({ ...articleEntity, isLiked });
+      }),
+    );
+
+    return {
+      ...articleListResult,
+      list: articles,
+    };
+  }
+
+  async postArticleComment(articleId: number, userId: number, content: CreateCommentRequest) {
+    await this.getExistingArticle(articleId);
+
+    const commentEntity = await this.commentRepository.createArticleComment(
+      articleId,
+      userId,
+      content,
+    );
+
+    return new Comment(commentEntity);
+  }
+
+  async getArticleComments(articleId: number, getCommentListDto: GetCommentListRequest) {
+    await this.getExistingArticle(articleId);
+
+    const getCommentListResult = await this.commentRepository.findComments({
+      ...getCommentListDto,
+      articleId,
+    });
+    const comments = getCommentListResult.comments.map((comment) => new Comment(comment));
```

Comment:
특별하게 대단한 기법은 아니지만, class의 static method를 잘 활용하면 이런 축약도 가능해집니다.

```suggestion
    const comments = getCommentListResult.comments.map(Comment.new);
```

이런 식으로 접근해볼 수도 있곘죠

```suggestion
    const comments = Comment.fromList(getCommentListResult.comments);
```

헌규님은 어떤 방식이 더 눈에 잘 들어오시나요?